Receiver/BMS_Receiver.cpp

```diff
@@ -0,0 +1,141 @@
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
+#include <boost/foreach.hpp>
+#include "BMS_Receiver.h"
+
+using boost::property_tree::ptree;
+
+
+
+void BMS_Receiver::get_data_from_console()
+{
+	string sender_data;
+	
+	for (int i = 0; i < 15;i++)
+	{
+		Parameters parameter;
+		std::getline(std::cin, sender_data);
+		
+		//cout<<sender_data<<endl;
+		if (parse_data(sender_data, parameter))
+		{
+			parameter_list.push_back(parameter);
+			calculate_statistics();			
+		}
+	}	
+}
+
+void BMS_Receiver::print_statistics()
+{
+	print_data();
+}
+
+void BMS_Receiver::calculate_statistics()
+{
+	perform_operation(parameter_list);		
```

Comment:
perform_operation doesn't convey the purpose of that function. Name it specifically - why should that function exist and why not call min, max and moving-avg directly from `calculate_statistics`?